### PR TITLE
Add API server.

### PR DESCRIPTION
**Pull request by bfabio**
Add a simple API server exposing the JSON data via HTTP.

The Dockerfile builds a small (~8MiB) stateless image (based on the data at `dati-json/`) which can be easily scaled up, fe. with Kubernetes replicas.

From DockerHub:
`$ docker run -p8000:8000 bfabio/covid-19`

**Edit:**
Temporarly available at https://covid-19.zland.org/

fe. https://covid-19.zland.org/regioni/Lombardia

**[API documentation](https://editor.swagger.io/?url=https://raw.githubusercontent.com/bfabio/COVID-19/api/api/openapi.yaml)**

### CODE REVIEW

**Comment by apiraino** [comment]
@bfabio if after the build you also add

```
RUN cd api && strip ./target/x86_64-unknown-linux-musl/release/api
```

the docker image size is halved :smile:

**Comment by bfabio** [comment]
@apiraino Good call

File: README.md [hunk 1]
```diff
@@ -137,6 +137,35 @@ COVID-19/
 
 *Viene messo a disposizione un file JSON complessivo di tutte le date nella cartella "dati-json": dpc-covid19-ita-andamento-nazionale.json*
 
+### API
+
+Tutte le chiamate accettano i parametri `begin` e `end` in formato ISO 8601.
```
Comment (ioggstream) [inline README.md]: Puoi fare una specifica OpenAPI anziché indicarle nel readme, eg: https://editor.swagger.io/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fteamdigitale%2Fapi-starter-kit%2Fmaster%2Fopenapi%2Fsimple.yaml

I can create a template for you.

This code could be interesting for creating an api-starter-kit-rust project like https://github.com/teamdigitale/api-starter-kit-python. WDYT

**Comment by macchie** [comment]
Well done. I was about to implement the same API Project on Loopback but you arrived first.

Thanks!

File: README.md (hunk 1 above)
Comment (bfabio) [reply README.md]: > Puoi fare una specifica OpenAPI anziché indicarle nel readme, eg: https://editor.swagger.io/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fteamdigitale%2Fapi-starter-kit%2Fmaster%2Fopenapi%2Fsimple.yaml

Sure. I think we should keep them in the README for the time being though, at least until we have a place to serve the rendered version.

I'll wait for the new format coming out today (see https://github.com/pcm-dpc/COVID-19/issues/115#issuecomment-597799832)

> This code could be interesting for creating an api-starter-kit-rust project like https://github.com/teamdigitale/api-starter-kit-python. WDYT

That's a great idea, but be aware: as other Rustaceans may have noticed, my Rust is pretty basic.

Comment (bfabio) [reply README.md]: @ioggstream WTH, OpenAPI [doesn't support optional path parameters yet](https://github.com/OAI/OpenAPI-Specification/issues/93). The documentation won't be 100% accurate.

Comment (bfabio) [reply README.md]: Looks like the format haven't change yet, so I added the spec anyway.

**Comment by bfabio** [comment]
> I can only suggest the minor change of keeping the Dockerfile in a sub-directory.

I think having it in the root directory it's useful: you notice straight away there's a Docker image you can build.

**Comment by umbros** [comment]
Ciao @bfabio sto testando una soluzione basata su lambda. Grazie per la collaborazione, puoi comunque pubblicare il docker su un tuo repo e fare riferimento a questo per i dati

**Pull request closed without merging**